include/zephyr/sys/async.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * @brief Public APIs for Async utilities
+ *
+ * Functions and datatypes useful for safely communicating to a thread or work
+ * queue of an asynchronous functions completion.
+ *
+ * All callback helpers here are safe to call from an ISR.
+ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_ASYNC_H_
+#define ZEPHYR_INCLUDE_SYS_ASYNC_H_
+
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
```

Comment:
note that content in this header doesn't seem to belong to any group, I think it should and at the same time be pulled in Sphinx side (with some usage examples maybe, like many Kernel APIs do)